CMakeLists.txt

```diff
@@ -463,6 +463,9 @@ endif()
 if(ENABLE_LIBICONV)
   find_package(Iconv REQUIRED)
   include_directories(SYSTEM ${Iconv_INCLUDE_DIRS})
+  if(MSVC)
+    set(CMAKE_REQUIRED_INCLUDES ${ICONV_INCLUDE_DIR})
```

Comment:
should this be:

    list(APPEND CMAKE_REQUIRED_INCLUDES "${ICONV_INCLUDE_DIR}")

based on:  https://github.com/neovim/neovim/blob/7086751c5e4eb3cfee0b98df0d3cedc8bff47d35/cmake/FindLibIntl.cmake#L33-L35

Maybe do it in `FindIconv.cmake`, without the MSVC check, similar to `FindLibIntl.cmake`?

```
 if (ICONV_INCLUDE_DIR) 
   list(APPEND CMAKE_REQUIRED_INCLUDES "${ICONV_INCLUDE_DIR}") 
 endif() 
```

Reply:
`iconv.h` was found in `FindIconv.cmake`, so I think the following changes are fine.
```diff
--- a/config/CMakeLists.txt
+++ b/config/CMakeLists.txt
@@ -20,7 +20,6 @@ endif()
 check_symbol_exists(_NSGetEnviron crt_externs.h HAVE__NSGETENVIRON)

 # Headers
-check_include_files(iconv.h HAVE_ICONV_H)
 check_include_files(langinfo.h HAVE_LANGINFO_H)
 check_include_files(locale.h HAVE_LOCALE_H)
 check_include_files(pwd.h HAVE_PWD_H)
@@ -43,6 +42,7 @@ check_function_exists(readv HAVE_READV)

 if(Iconv_FOUND)
   set(HAVE_ICONV 1)
+  set(HAVE_ICONV_H 1)
 endif()

 check_function_exists(opendir HAVE_OPENDIR)
```

Reply:
ok, and I see that #10708 removes `HAVE_ICONV_H`.

Reply:
#10708 makes this unnecessary. However, this change is safer than #10708 and is left open.